File: back-end/Domain/src/Models/Instituicao/InstituicaoCursoOcorrenciaProfessor.cs
using System;
using System.ComponentModel.DataAnnotations;
using Domain.Models.Interface;

namespace Domain.Models {
    public class InstituicaoCursoOcorrenciaProfessor : IBaseModel {

        public InstituicaoCursoOcorrenciaProfessor() {

        }

        [Key]
        public long ID { get; set; }
        public InstituicaoCursoOcorrencia InstituicaoCursoOcorrencia { get; set; }
        public Professor Professor { get; set; }
        public InstituicaoCursoPeriodo Periodo { get; set; }
        public CursoGradeMateria Materia { get; set; }
        public InstituicaoCursoTurma Turma { get; set; }
        public bool Confirmado { get; set; }

        public DateTime? DataInicio { get; set; }
        public DateTime? DataExpiracao { get; set; }

        public bool Ativo { get; set; } = true;

    }
}

File: front-end/src/util/factory/instituicao/instituicao-curso-ocorrencia-professor.ts
import { Professor } from '../usuario/professor';
import { InstituicaoCursoTurma } from './instituicao-curso-turma';
import { InstituicaoCursoPeriodo } from './instituicao-curso-periodo';
import { CursoGradeMateria } from '../curso/curso-grade-materia';

export class InstituicaoCursoOcorrenciaProfessor {
    id ?: string;
    professor: Professor;
    cursoGradeMateria: CursoGradeMateria;
    turma: InstituicaoCursoTurma;
    periodo: InstituicaoCursoPeriodo;
}